lib/crons.ex

```diff
@@ -0,0 +1,135 @@
+defmodule ApplicationRunner.Crons do
+  @moduledoc """
+    ApplicationRunner.Crons delegates methods to the corresponding service.
+  """
+
+  import Ecto.Query, only: [from: 2, from: 1]
+
+  alias ApplicationRunner.Crons.Cron
+  alias ApplicationRunner.Errors.{BusinessError, TechnicalError}
+  alias ApplicationRunner.{AppSocket, Environment, EventHandler, Repo}
+  alias Crontab.CronExpression.{Composer, Parser}
+
+  def run_env_cron(
+        action,
+        props,
+        event,
+        env_id,
+        function_name
+      ) do
+    with {:ok, token} <-
+           AppSocket.do_create_env_token(env_id),
+         {:ok, _pid} <-
+           Environment.ensure_env_started(%Environment.Metadata{
+             env_id: env_id,
+             function_name: function_name,
+             token: token
+           }) do
+      EventHandler.send_env_event(env_id, action, props, event)
+    end
+  end
+
+  def create(env_id, function_name, %{"listener_name" => _action} = params) do
```

Comment:
Add with/case/if to check if the cron is created before adding the task and the return name.